src/lib/registration/ui/abort_confirmation.rb

```diff
@@ -0,0 +1,42 @@
+require "yast"
+
+module Registration
+  module UI
+    # this class displays a popup for the user to confirm the abort
+    class AbortConfirmation
+      Yast.import "Popup"
+      Yast.import "Mode"
+
+      # displays a new popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def self.run
+        dialog = new
+        dialog.run
+      end
+
+      # the constructor
+      def initialize
+        textdomain "registration"
+      end
+
+      # displays the popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def run
+        if Yast::Mode.installation || Yast::Mode.repair
+          Yast::Popup.ConfirmAbort(:painless)
+        else
+          msg = _("Really abort the registration?\n\n" \
```

Comment:
I'd probably prefer simple "Rally abort?" question with `[Yes][No]` buttons.

- The `no changes will be done to the system` statement is not completely true, registration workflow has several steps, e.g. you register the base product, then can optionally register and install an extension. But when you abort the extension selection dialog, the system has been already registered (=changed)

- `Really abort the *registration*?` is also not completely correct, some registration parts are called from the online migration module, distinguishing regular registration or online migration would make it complex.

Therefore just the simple "Rally abort?" is the best IMHO...